types/react-test-renderer/index.d.ts

```diff
@@ -25,7 +25,7 @@ export interface ReactTestRendererTree extends ReactTestRendererJSON {
 }
 export interface ReactTestInstance {
     instance: any;
-    type: string;
+    type: any;
```

Comment:
Could this be `ReactType` instead? That would make it symmetrical with `findByType` and `findAllByType`.

Reply:
I thought about that, but the problem is that `findByType` and `findAllByType` are wrong too, based on:
https://github.com/facebook/react/blob/v16.0.0/src/renderers/testing/ReactTestRendererFiberEntry.js#L493
It assumes that it can be `any`.

If react-test-renderer typings are fixed to remove these `any` and point to `ReactType`, then I would agree, but even on the latest version https://github.com/facebook/react/blob/master/packages/react-test-renderer/src/ReactTestRenderer.js#L335 they have `any`.
I'm just trying to make it the same as flow definitions.

I can also fix `findByType` and `findAllByType` defs in this PR if you prefer.
Thoughts?

Reply:
I would suspect that the use of `any` in this part of the React codebase may be to avoid creating a dependency from a public interface to a private interface (or something like that). In other words, I don't believe they actually mean it could be _any_ type.

IMO, it's more important to be consistent within the React typings, rather than being 100% true to React internals. Since the `type` property on the public `ReactElement` interface uses `ReactType`, it would make sense (to me) to use `ReactType` here as well.

Reply:
@jgoz, made the suggested change.